src/awkward/_errors.py

```diff
@@ -31,6 +32,17 @@ def __call__(self):
         return self.func(*self.args, **self.kwargs)
 
 
+class AwkwardKeyError(builtins.KeyError):
+    def __str__(self):
+        return super(Exception, self).__str__()
+
+
+# Pretend to be builtins!
+AwkwardKeyError.__name__ = "KeyError"
+AwkwardKeyError.__qualname__ = "KeyError"
+AwkwardKeyError.__module__ = "builtins"
```

Comment:
I'm wary of this. It could lead to someone being very confused that this type `is not KeyError` for some reason.

I was thinking that we would name the class `ak._errors.KeyError`, so it displays its name as `KeyError`, but it's distinguishable by its `__module__`. Setting `__module__ = "builtins"` sounds like we're setting up for a disaster.

Reply:
That's fair. In general, it's not good practice to use 'is' like this, but I think you make a reasonable argument. Let's rework this in line with making our error distinguishable from the built-in.